scripts/lib/CIME/namelist.py

```diff
@@ -912,38 +912,38 @@ def get_variable_names(self, group_name):
         []
         >>> x = parse(text='&foo bar=,bazz=true,bazz(2)=fred,bang=6*""/')
         >>> sorted(x.get_variable_names('fOo'))
-        [u'bang', u'bar', u'bazz', u'bazz(2)']
+        ['bang', 'bar', 'bazz', 'bazz(2)']
         >>> x = parse(text='&foo bar=,bazz=true,bang=6*""/')
         >>> sorted(x.get_variable_names('fOo'))
-        [u'bang', u'bar', u'bazz']
+        ['bang', 'bar', 'bazz']
         >>> x = parse(text='&foo bar(::)=,bazz=false,bazz(2)=true,bazz(:2:)=6*""/')
         >>> sorted(x.get_variable_names('fOo'))
-        [u'bar(::)', u'bazz', u'bazz(2)', u'bazz(:2:)']
+        ['bar(::)', 'bazz', 'bazz(2)', 'bazz(:2:)']
         """
         group_name = group_name.lower()
         if group_name not in self._groups:
             return []
-        return self._groups[group_name].keys()
+        return list(self._groups[group_name].keys())
```

Comment:
I'm wondering if the conversion to `list` here was done just to get the doctests to pass? If so, I'd say a better solution would be to convert the result to `list` in the doctests - keeping the production code returning an iterator in python3.